front/src/views/Transfers/CreateTransferAgreement/components/CreateTransferAgreement.tsx

```diff
@@ -174,6 +177,8 @@ function CreateTransferAgreement({
     validFrom,
   ]);
 
+  if (isGlobalStateLoading) return <APILoadingIndicator />;
```

Comment:
is there a reason why you put this in CreateTransferAgreement and not in the container component CreateTransferAgreementView? Otherwise, I'd move it up.